src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -867,6 +867,61 @@ LogicalResult ONNXSeluOp::inferShapes(
   return success();
 }
 
+LogicalResult ONNXSequenceInsertOp::inferShapes(
+    std::function<void(mlir::Region &)> doShapeInference) {
+  return success();
+}
+
+static LogicalResult verify(ONNXSequenceInsertOp op) {
+  ONNXSequenceInsertOpAdaptor operandAdaptor = ONNXSequenceInsertOpAdaptor(op);
+
+  // These cast should be guaranteed by default verifier
+  auto seqElementType = operandAdaptor.input_sequence()
+                            .getType()
+                            .dyn_cast<mlir::onnxmlir::SeqType>()
+                            .getElementType();
+  auto elementType1 = seqElementType.dyn_cast<ShapedType>().getElementType();
+  auto insertType = operandAdaptor.tensor().getType().dyn_cast<ShapedType>();
```

Comment:
Same here, don't we have to test if we don't have a shape type? See the patterns in the other verifiers

Reply:
```
ShapedType is a common base class between Vector, UnrankedTensor, RankedTensor, and MemRef types because they share behavior and semantics around shape, rank, and fixed element type
```
All the input and output of operations in onnx-mlir dialect should be shapedType except for seq type. The default verifier should guarantee  that. This may not be true after lowering.